src/lando/utils/auth.py

```diff
@@ -60,6 +63,24 @@ def authenticate(self, request: WSGIRequest, token: str) -> User:
         return request.user
 
 
+class PermissionAccessTokenAuth(AccessTokenAuth):
+    """Ninja authenticator which also verifies that the user has a given permission."""
+
+    required_permission: str
+
+    def __init__(self, required_permission: str, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.required_permission = required_permission
+
+    @override
+    def authenticate(self, request: WSGIRequest, token: str) -> User:
+        user = super().authenticate(request, token)
+        # XXX Check the user's own permissions, not delegated from other roles.
```

Comment:
What does this comment mean? We don't currently leverage Django permission groups, but if we did start using them this would still be okay, no?